### PR TITLE
K8s update mar 2023

### PR DESCRIPTION
**Pull request by godber**
I had this working with the example job with the `initContainers` added, that ran and everything so it worked.  I've made some changes since then so it's possible that I broke it again, but its close.  I need to add tests and implement the `kubernetes_overrides_enabled` config setting

refs: 3347

### CODE REVIEW

**Comment by godber** [comment]
I think that I should reject a job a job submission time if `kubernetes_overrides_enabled: false` and the job has a non null (`{}`) `pod_spec_override`.  Can I do that in `format` in `job-schemas.ts`:

https://github.com/terascope/teraslice/pull/3348/files#diff-501767f567d0fd519fa00553a9d316cef76c0675f54f2a3443ccda88951af053R314

Or is there  job validation elsewhere, I feel like there was other job validation.

File: docs/configuration/clustering-k8s.md [hunk 1]
```diff
@@ -436,6 +437,39 @@ property should be the name of the Kubernetes `persistentVolumeClaim`.  The
...
+
+Teraslice provides the `pod_spec_override` job property which allows the user to
+specify any arbitrary Kubernetes configuration to be overlaid on the
+[pod](https://kubernetes.io/docs/concepts/workloads/pods/) `.spec` for both the
+execution controller and worker pods.  This allows users to "sneak" in
+functionality that we have not yet explicitly implemented.
+
+For instance, this feature can be used to add `initContainers` or `hostAliases`.
+A `pod_spec_override` must be in the form of JSON that maps directly to the
+appropriate YAML for the Kubernetes resource.  And example of a
```
Comment (ciorg) [inline docs/configuration/clustering-k8s.md:450]: Typo - `An example`

**Review by ciorg — APPROVED** [review]
Approving, found one small typo

File: docs/configuration/clustering-k8s.md (hunk 1 above)
Comment (godber) [inline docs/configuration/clustering-k8s.md:450]:
```suggestion
appropriate YAML for the Kubernetes resource.  An example of a
```

Comment (godber) [reply docs/configuration/clustering-k8s.md]: Thanks, I've got that fixed, I merge when it completes.

**Pull request merged**